server-side/my.service.ts

```
@@ -560,21 +548,49 @@ export class DataService {
         };
     }
 
-    // get all table documents in parallel
-    async prepareUDCReport(archiveData: ScheduledUDC[], report: ReportTuple[]): Promise<ScheduledUDC[]> { 
-        const deletedTables: Set<string> = new Set();
-        
-        await async.mapLimit(archiveData,
-            PARALLEL_LIMIT,
-            async (table: ScheduledUDC) => {
-            try{
-                const tableResult = await this.processResource(table);
-                report.push(tableResult);
-                deletedTables.add(table.Name!);
-            }  catch(err) {
-                console.error(`Error while preparing UDC report for table ${table.Name}, error: ${err}`);
+   CheckTime(startDate: number): boolean {
+    const remainingTime = (LAMBDA_MAX_EXECUTION_TIME - (Date.now() - startDate));
+    return remainingTime > MIN_RESOURCE_EXECUTION_TIME;
+ }
+
+ async parallelExecution(startDate: number, archiveData: ScheduledUDC[], report: ReportTuple[], resourcesToProcess: string[]): Promise<void> {
+    await async.mapLimit(archiveData,
+        PARALLEL_LIMIT,
+        async (table: ScheduledUDC) => {
+            if(this.CheckTime(startDate)) {
+                try{
+                    const tableResult = await this.processResource(table);
+                    report.push(tableResult);
+                    resourcesToProcess.push(table.Name!);
+                }  catch(err) {
+                    console.error(`Error while preparing UDC report for table ${table.Name}, error: ${err}`);
+                }
+            } else {
+                throw new Error('Time limit reached'); // in order to exit mapLimit at the current iteration
```

why are you throwing an error? why not just return the function.. the retry will take care of it
I don't think this error is not propagated to the caller. maybe just write a warning to the consol